aws_lambda_builders/workflows/java_gradle/gradle_validator.py

```diff
@@ -0,0 +1,60 @@
+"""
+Gradle Binary Validation
+"""
+
+import logging
+import re
+
+from aws_lambda_builders.exceptions import LambdaBuilderError
+from .utils import OSUtils
+
+LOG = logging.getLogger(__name__)
+
+
+class GradleBinaryValidatorError(LambdaBuilderError):
+    MESSAGE = "gradle executable found failed to return version"
+
+
+class GradleBinaryValidator(object):
+    MAJOR_VERSION_WARNING = "%s is using a JVM with major version %s which is newer than 8 that is supported by AWS " \
+                            "Lambda. The compiled function code may not run in AWS Lambda unless the project has " \
+                            "been configured to be compatible with Java 8 using 'targetCompatibility' in Gradle."
+
+    def __init__(self, os_utils=None, log=None):
+        self.language = 'java'
+        self._valid_binary_path = None
+        self.os_utils = OSUtils() if not os_utils else os_utils
+        self.log = LOG if not log else log
```

Comment:
this is not a good pattern in Python. using LOG is the way to go.

Reply:
This was simply so I could test that the validator was actually logging the warning (by passing in a mock log object). This is the pattern we use in the Java SDK. Is there a more idiomatic way to test this in Python?

Reply:
Makes sense. Usually you don't test for log statements, but in this case it makes sense.